### PR TITLE
fix: Enforce verification key integrity check regardless of TRUST_PARAMS=1

### LINKED ISSUE #6640: [BUG] `lotus daemon` should refuse to run unless all verification params are present
**Describe the bug**

During the recent upgrade a number of folks ended up running their nodes with the new aggregate-proof validation param missing. This led to the formation of an uncomfortably long-lived side-chain.

Since the validation set is really small ( less than 300MiB ), we should just check its presence and integrity on every startup, regardless of `TRUST_PARAMS` or any other escape hatches.

### PR DESCRIPTION
**Pull request by placer14**
closes #6640 

See change in https://github.com/filecoin-project/go-paramfetch/pull/17 for details.

### CODE REVIEW

**Comment by placer14** [comment]
Please check that I did the module dependency dance properly. A `go mod tidy` seemed to reverse the update to `go-fetchparam` which was not what I expected (as it is neither unused or missing). 🤷

**Comment by placer14** [comment]
ATTENTION: HOLD THIS PR FROM BEING MERGED!

(I realize that the dependent package may squash-merge causing the final commit SHA to change. This would cause the build to break later once dangling branches are garbage-collected and that would be bad.)

**Comment by placer14** [comment]
Holding this PR as I now realize the dependent package could be squash-merged causing the commit SHA to change. I'll ready this again once the dependent PR is merged and SHA updated.

File: go.mod
Comment (magik6k) [inline go.mod:42]: Nit: use a tagged version

**Review by magik6k — APPROVED** [review]

**Pull request merged**
